image/image.go

```diff
@@ -102,6 +102,36 @@ func localSnaps(opts *Options) (*localInfos, error) {
 			info.Revision = snap.R(-1)
 			nameToPath[info.Name()] = snapName
 			local[snapName] = info
+
+			// crude way to find the .assert file
+			assertFile := strings.Replace(snapName, ".snap", ".assert", -1)
+			// find the revision/snap-id from the local provided
+			// assertion and set info accordingly. no need to
+			// actually check the assertion because we do that
+			// anyway for all snaps that have a snapID/revision.
+			f, err := os.Open(assertFile)
+			if os.IsNotExist(err) {
+				continue
+			}
+			if err != nil {
+				return nil, err
+			}
+			defer f.Close()
+			dec := asserts.NewDecoder(f)
+			for {
+				a, err := dec.Decode()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return nil, err
+				}
+				if sr, ok := a.(*asserts.SnapRevision); ok {
```

Comment:
Later as in later when you are online or when some other process re-processes the same assertion files?